cpp/src/parquet/stream_writer.cc

```diff
@@ -183,6 +187,28 @@ StreamWriter& StreamWriter::WriteFixedLength(const char* data_ptr, std::size_t d
   return *this;
 }
 
+StreamWriter& StreamWriter::WriteRawDataVariableLength(const uint8_t* data_ptr,
```

Comment:
This is almost exactly the same code as in `WriteVariableLength`, right? Can you refactor these methods to share the common parts? (only `CheckColumn` and the pointer cast seem to differ...)

Reply:
Done, I hope changing a protected API is not a problem as I added a new parameter to WriteVariableLength. I don't know what your rules for API/ABI stability are.